### PR TITLE
Odpm 66 additional endpoints

### PR DESCRIPTION
**Pull request by rlconley**
This pr adds the following endpoints for completing the MD graphs. The data structure should match that for nc:

/md/api/agency/,id./searches/
/md/api/agency/,id./stops_by_reason/
/md/api/agency/,id./contraband_hit_rate/
provides number of searches and number of times contraband was seized per year and race

All of these endpoints are viewable in the browser

### CODE REVIEW

File: md/api.py [hunk 1]
```diff
@@ -58,5 +60,44 @@ def stops(self, request, pk=None):
...
+    @detail_route(methods=['get'])
+    @cache_response(key_func=query_cache_key_func)
+    def contraband_hit_rate(self, request, pk=None):
+        response = {}
+        # searches
+        results = GroupedData(by='year', defaults=GROUP_DEFAULTS)
+        q = Q(search_conducted='Y')
+        self.query(results, group_by=('year', 'ethnicity'), filter_=q)
+        response['searches'] = results.flatten()
+        # searches
```
Comment (trawick) [inline md/api.py]: I suspect you meant that comment to say "contraband".

File: md/api.py [hunk 2]
```diff
@@ -10,11 +10,10 @@
...
 from md import serializers
-from md.models import ETHNICITY_CHOICES
+from md.models import ETHNICITY_CHOICES, PURPOSE_CHOICES
 from tsdata.utils import GroupedData
 
 
-# PURPOSE_CHOICES to be added after ODPM-31
 GROUP_DEFAULTS = {k: 0 for k in dict(ETHNICITY_CHOICES).values()}
 
 class QueryKeyConstructor(DefaultObjectKeyConstructor):
```
Comment (trawick) [inline md/api.py]: existing code, but add another blank line before the class for PEP 8 happiness

File: md/api.py [hunk 3]
```diff
@@ -58,5 +60,44 @@ def stops(self, request, pk=None):
...
+        response['searches'] = results.flatten()
+        # searches
+        results = GroupedData(by='year', defaults=GROUP_DEFAULTS)
+        q = Q(seized='Y')
+        self.query(results, group_by=('year', 'ethnicity'), filter_=q)
+        response['contraband'] = results.flatten()
+        return Response(response)
+
     # for additional methods related to searches, look first at most recent
     # corresponding nc code
```
Comment (trawick) [inline md/api.py]: This existing comment doesn't apply anymore, right?

File: md/api.py [hunk 4]
```diff
@@ -44,7 +43,10 @@ def query(self, results, group_by, filter_=None):
             data = {}
             if 'year' in group_by:
                 data['year'] = stop['year'].year
-            # XXX check for 'purpose' here after ODPM-31 is delivered.
+            if 'purpose' in group_by:
+                purpose = dict(PURPOSE_CHOICES).get(stop['purpose'],
+                                              stop['purpose'])
```
Comment (trawick) [inline md/api.py]: this code should line up with something, such as the stop['purpose'] on the line above  (similarly for the existing code a few lines below)

**Comment by trawick** [comment]
I noticed that stops aren't getting assigned to years by the API exactly like they are grouped by normal Django ORM queries, which could explain the rare stop showing up in 2016.  I added a note about the suspected cause to ODPM-46.  (not a concern for this ticket)

**Comment by trawick** [comment]
Conditional +1 :)

1. Address the trivial comments I made (not the year issue related to ODPM-46)
2. Get Neil to look at the responses of the new API endpoints and confirm that the front-end code can use it as easily as it uses the corresponding NC API responses.

**Comment by copelco** [comment]
If we don't include tests in this PR, will they be handled in another ticket this sprint?

File: md/api.py (hunk 2 above)
Comment (rlconley) [inline md/api.py]: done

File: md/api.py (hunk 3 above)
Comment (rlconley) [inline md/api.py]: done

File: md/api.py (hunk 4 above)
Comment (rlconley) [inline md/api.py]: done

**Comment by nmashton** [comment]
AFAIC, :shipit:

**Pull request merged**